### PR TITLE
Review Google Authenticator int guide

### PR DESCRIPTION
**Pull request by robertnicolo-okta**
## Notice to reviewers

This particular PR is meant to be a place for feedback and comments for the Google Authenticator Guide for the Javascript SDK located at the following deploy link: https://61fa99ab0c9964008b14ee42--reverent-murdock-829d24.netlify.app/docs/guides/authenticators-google-authenticator/nodeexpress/main/

It doesn't include the actual changes.  Review the guide located at the deploy link and place your comments here in this PR. Thanks!

## PR Description:
- Add the google authentication integration guide. 

### Resolves:

* [OKTA-459224](https://oktainc.atlassian.net/browse/OKTA-459224)

### CODE REVIEW

**Review by chrisdmills-okta — COMMENTED** [review]
This is looking pretty good @robertnicolo-okta !

Some comments for you:

* For related use cases, I'm thinking that we should have a separate "Related use cases" section to list those, rather than including those links in the opening paragraph. I'm not sure if that should live in the nut facts, say in the sample code sectio? I'm thinking probably not, so maybe just below that?

* Do the "Integrate SDK for authenticator enrollment" and "Integrate SDK for authenticator challenge" sections need flow diagrams to go along with the enrollment flow steps?

* The final images should be done by Alex Cohen, for consistency with the image designs he presented to us recently. You should open us a design ticket to ask him to help with these.

* I was expecting some kind of a short section at the top of the document that introduces the authenticator in each case - what you can do with it, what the experience is like for the user. Nothing too long, but we shouldn't assume that the reader knows exactly what the authenticator is and what they can do with it. 

Something like:

## Google authenticator overview

Google authenticator is an app available for a variety of platforms that can provide an extra security factor for your apps.

[SCREENSHOT]

1. The user is expected to install the app and sign into it using their Google account credentials.
2. Once they are signed in, they can generate a verification code that is then entered into the associated app.

Or somesuch thing (probably inaccurate, but I did come up with it off the top of my head)

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/index.md [hunk 1]
```diff
@@ -4,3 +4,5 @@ layout: Guides
 sections:
  - main
 ---
+
+<!--File needs review-->
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/index.md:8]: Where is all of the content to be reviewed? Is the Netlify link the only way to review the content? Thanks!

Additionally - our heading style standard is to not number headings. what is the reasoning behind this? Thank you!

**Review by susanharper-okta — COMMENTED** [review]
a couple of questions. thanks.

**Review by danielmaharry-okta — CHANGES_REQUESTED** [review]
**index.md**
Line 12 : Need to change the example scenarios that GAuth can be used for. Password-only and Facebook-only are your two examples and they don't count.
Line 55 : Clicking Device Bound isn't necessary
Line 56 : Spelling mistake - "requirement" not "requirment"

**challengeintegrationsteps.md**
Line 3 : Rather than pointing to the basic user sign-in flow, this is MFA isn't it, so we probably need to create a generic "Sign In With Password and Authenticator" guide which we can point all three specific authenticator guides to. The point is, once we add GAuth, it isn't basic sign-in flow any more.
Line 14 : The section is called "Update configurations" not "Configuration updates"

**enrollmentintegrationsteps.md**
Line 3 : Rather than pointing to the basic user sign-in flow, this is MFA isn't it, _**so we probably need to create a generic "Sign In With Password and Authenticator" guide which we can point all three specific authenticator guides to.**_ The point is, once we add GAuth, it isn't basic sign-in flow any more.
Line 14 : The section is called "Update configurations" not "Configuration updates"
Line 53 : There's a spelling mistake in the callout in the PNG file. Should be "originates", not "orignates"

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/index.md (hunk 1 above)
Comment (robertnicolo-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/index.md:8]: Yes that's right Susan. I'll sync up with you to discuss other review options.

Comment (robertnicolo-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/index.md:8]: Numbers removed.

**Review by shuowu-okta — APPROVED** [review]
Code snippets and diagrams look good to me!

**Review by susanharper-okta — COMMENTED** [review]
Few global notes to address that I don't repeat in my suggested edits:
 
-Dialog box = dialog (per InfoDev and Microsoft)
-It's In (not On) a dialog, On a page, and On a tab.
-Id token -> ID token
-Application -> app (InfoDev - line 9: "In general, use instead of application. You may need to use application to refer to the UI or to formalize a particular 3rd-party application in a process.") 

-----
Learning outcomes 

Second bullet needs a period at the end.

Need a space between the Sample code URL and the title "Related use cases"
-----
Overview

A few changes to the first two sentences:

Google Authenticator is an authenticator app developed by Google that is used to verify the identity of a user. During authentication (for example, sign-in), a user that is enrolled in Google Authenticator is challenged by the service provider to provide a Time-based One-Time Password (TOTP). 

Add "of" to first sentence:

The TOTP is calculated based off of the current time and a shared key that was generated by the service provider and added to the Google Authenticator app during initial authenticator enrollment. 

Cap Authenticator:

The following diagram illustrates how the Google Authenticator enrollment and challenge flows can work in your app.

-----
Update configurations

I notice that you use both Okta org application and app integration - we should just use one reference (preferably app integration - since that is the standard).

Before you can start using the Google Authenticator, create an app integration in Okta as described in.......

-----
Configure your Okta org application to use Google Authenticator -> Configure your app integration to use Google Authenticator

Step two - remove "on", you've -> that you

On the Applications page, click the app that you previously created.

Step five - use either bullets for the sub-steps or letters.

Sub-step 4 - adding THAT, satisfy and requirement are misspelled, and it's to, not too

Confirm that **Your org's authenticators that satisfy this requirement** is set to **Password AND Google Authenticator or ....**
-----

Install Google Authenticator

Extra "either"

Install the Google Authenticator app on your mobile device using either the Google Play Store (Android) or Apple App Store (iOS).

-----
Integrate SDK for authenticator enrollment

Summary of steps - adding the subject to the sentence = CHART and summarizes not summaries.

The following chart summarizes the Google Authenticator enrollment flow using a basic user sign-in use case.

-----
Initiate use case requiring authentication -> Initiate use case that requires authentication

Same change for first sentence

Initiate a use case that requires authentication.

-----
Display Google Authenticator option

Present tense change to first sentence and adding the word "the":

If you configure your Okta org as detailed in the Configuration updates, authenticate() returns a response with Google Authenticator in the list of available authenticators. 

Adding IS after nextStep.name and Google Authenticator in second sentence

Specifically, IdxTransaction is returned with a status of PENDING, nextStep.name IS set to select-authenticator-enroll, and Google Authenticator IS included as an option in the nextStep.options array.

Sentence after IdxTransaction example - couple things:

Use the value and label properties to show the available list of authenticators to the user. The sample app constructs a drop-down box using a Mustache template.

-----
Submit Google Authenticator option

When the user selects and submits Google Authenticator, call OktaAuth.idx.proceed() and pass in the google_otp value from IdxOption.value.

-----
Display shared secret and QR code

A "that" edit, a "that contains" edit, a "that stores" edit, and removal of "for more details"

Next, display the shared secret to the user so that they can copy the value to the Google Authenticator app. The response from OktaAuth.idx.proceed() allows you to display a string and QR code that contains the shared secret. IdxTransaction is returned with authenticator.contextualData.sharedsecret set to a string of the secret and authenticator.contextualData.qrcode.href that stores the secret in a base64-encoded PNG image. See the following IdxTransaction example.

Intro sentence for Enroll Google Authenticator: dropdown list -> drop-down list:

-----
Copy shared secret to Google Authenticator

First sentence: is displayed -> appears

After the shared secret appears, the user installs the Google Authenticator app on their mobile device if it's not already installed. 

Last sentence - Once is difficult to translate..use After...try this? Also - use the abbreviation that you introduced! :)

After the secret code is added, Google Authenticator displays the TOTP for the newly added account.

-----
Submit one-time password in your app

Is this the TOTP? If so, can we not continue to just use the abbreviation? Is it the same thing or different? I may be not understanding.....if I'm not, then ignore the rest of my comments about it following...haha

Once -> After
Passing in -> and pass in....

Submit TOTP in your app

The user then copies the TOTP from .......After the user submits.....call OktaAuth.idx.proceed() and pass in the password.

-----
Integrate SDK for authenticator challenge

Summary of steps

Adding in subject and summaries -> summarizes

The following chart summarizes the Google Authenticator challenge flow using a basic user sign-in use case.

-----
Initiate use case requiring authentication

requiring -> that requires

The first step is to initiate a use case that requires authentication. 

-----
Display Google Authenticator challenge

you've configure -> you configure, add "the" before Configuration updates, has already been enrolled -> is enrolled, indicating that, adding "is", removed "for more details"

If you configure your Okta org as detailed in the Configuration updates and the Google Authenticator is enrolled for the user, authenticate() returns a response indicating that  Google Authenticator is required for verification. Specifically, IdxTransaction is returned with a status of PENDING, nextStep.name is set to challenge-authenticator, and nextStep.authenticator is filled with Google Authenticator details. See the following IdxTransaction example.

Sentence after example - several edits:

Use the IdxTransaction properties to show a challenge page that displays a TOTP input field. The sample app constructs this page using the Mustache template.

Intro to Challenge Google Authenticator sentence should use "TOTP input field" rather than "one-term password input field". Right?

-----
Get one-time password from Google authenticator -> Get TOTP from Google Authenticator

A in authenticator, TOTP rather than one-time password, their -> the

Next, the user opens Google Authenticator on their mobile device and finds the TOTP for their account.

-----
Submit one-time password in your app -> Submit TOTP in your app

one-time password -> TOTP, passing in -> and pass in

When the user enters and submits this TOTP, call OktaAuth.idx.proceed() and pass in the password.

**Pull request closed without merging**